janus/src/hpke.rs

```diff
@@ -330,24 +341,17 @@ mod tests {
 
     #[test]
     fn round_trip_all_algorithms() {
-        round_trip_check::<DhP256HkdfSha256, HkdfSha256, AesGcm128>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha256, AesGcm256>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha256, ChaCha20Poly1305>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha384, AesGcm128>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha384, AesGcm256>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha384, ChaCha20Poly1305>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha512, AesGcm128>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha512, AesGcm256>();
-        round_trip_check::<DhP256HkdfSha256, HkdfSha512, ChaCha20Poly1305>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha256, AesGcm128>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha256, AesGcm256>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha256, ChaCha20Poly1305>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha384, AesGcm128>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha384, AesGcm256>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha384, ChaCha20Poly1305>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha512, AesGcm128>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha512, AesGcm256>();
-        round_trip_check::<X25519HkdfSha256, HkdfSha512, ChaCha20Poly1305>();
+        for kem_id in [HpkeKemId::P256HkdfSha256, HpkeKemId::X25519HkdfSha256] {
```

Comment:
This could potentially be `hpke_dispatch::KEM_ALL` (and likewise `KDF_ALL`, and `AEAD_ALL`) but I don't have a good sense of whether it's actually an advantage to have the explicit reenumeration here

Reply:
The idea here is to enumerate over `janus` representations of the KDF, KEM and AEADs. It's possible this test is entirely redundant if `hpke-dispatch` already exercises all these cases.

Reply:
This test is kinda redundant with `tests/round_trip.rs` in `hpke-dispatch`, except that this does exercise the Janus representation of stuff like algorithm IDs and HPKE configurations, so let's keep it.